extensions/eclipse-che-theia-prefs-provider/src/browser/prefs-provider.ts

```diff
@@ -0,0 +1,130 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+import {
+    PreferenceServiceImpl,
+    PreferenceScope,
+    FrontendApplicationContribution,
+    FrontendApplication
+} from '@theia/core/lib/browser';
+import { WorkspaceService } from '@theia/workspace/lib/browser';
+import WorkspaceClient, { IRemoteAPI } from '@eclipse-che/workspace-client';
+import { che } from '@eclipse-che/api';
+
+export interface WorkspaceWithDevfile extends che.workspace.Workspace {
```

Comment:
I'm curious, why do we need to extend che.workspace.Workspace ? devFile should already be there no ?

Reply:
I've just forgot to remove this. There was a type guard for a workspace with devfile, but later I got rid of it.